packages/datadog-plugin-mocha/src/index.js

```diff
@@ -0,0 +1,141 @@
+const { promisify } = require('util')
+
+const id = require('../../dd-trace/src/id')
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+const { SAMPLING_PRIORITY } = require('../../../ext/tags')
+const { AUTO_KEEP } = require('../../../ext/priority')
+const { getGitMetadata } = require('../../dd-trace/src/plugins/util/git')
+const { getCIMetadata } = require('../../dd-trace/src/plugins/util/ci')
+const {
+  TEST_FRAMEWORK,
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS
+} = require('../../dd-trace/src/plugins/util/test')
+
+const SPAN_TYPE = 'span.type'
+const RESOURCE_NAME = 'resource.name'
+
+function getTestMetadata () {
+  // TODO: eventually these will come from the tracer (generally available)
+  const ciMetadata = getCIMetadata()
+  const gitMetadata = getGitMetadata()
+
+  return {
+    [TEST_FRAMEWORK]: 'mocha',
+    ...ciMetadata,
+    ...gitMetadata
+  }
+}
+
+function createWrapRunTest (tracer, testMetadata) {
+  return function wrapRunTest (runTest) {
+    return async function runTestWithTrace () {
+      const childOf = tracer.extract('text_map', {
+        'x-datadog-trace-id': id().toString(10),
+        'x-datadog-parent-id': '0000000000000000',
+        'x-datadog-sampled': 1
+      })
+      const { file: testSuite, title: testName } = this.test
+      let specFunction = this.test.fn
+
+      if (specFunction.length) {
+        specFunction = promisify(specFunction)
+        // otherwise you have to explicitly call done()
+        this.test.async = 0
+        this.test.sync = true
+      }
+
+      this.test.fn = tracer.wrap(
+        'mocha.test',
+        {
+          type: 'test',
+          childOf,
+          resource: this.test.fullTitle(),
+          tags: {
+            [TEST_TYPE]: 'test',
+            [TEST_NAME]: testName,
+            [TEST_SUITE]: testSuite,
+            [SAMPLING_RULE_DECISION]: 1,
+            [SAMPLING_PRIORITY]: AUTO_KEEP,
+            ...testMetadata
+          }
+        },
+        async () => {
+          let result
+          try {
+            result = await specFunction()
+            tracer.scope().active().setTag(TEST_STATUS, 'pass')
+          } catch (error) {
+            tracer.scope().active().setTag(TEST_STATUS, 'fail')
+            throw error
+          } finally {
+            tracer
+              .scope()
+              .active()
```

Comment:
It would be cleaner to assign a variable to `tracer.scope().active()` since it is used multiple times in that function, especially when spanning multiple lines.

Reply:
addressed in https://github.com/DataDog/dd-trace-js/pull/1162/commits/4a6e603b75ca0e34c468cda9a31c41cf1b603b04#diff-3e917ea1366e30454d0aa44c11e64378545e1643def0c66baea963417518bc0dR76-R85